youroom/static/js/publicacion.js

```diff
@@ -0,0 +1,155 @@
+class Etiqueta {
+    constructor(nombre, enlace, coordX, coordY) {
+        this.nombre = nombre;
+        this.enlace = enlace;
+        this.coordX = coordX;
+        this.coordY = coordY;
+    }
+
+    set addEnlace(enlace) {
+        this.enlace = enlace;
+    }
+}
+
+let contadorEtiqueta = 0;
+let etiqueta = null;
+let listaEtiquetas = [];
+
+function procesarEtiquetas(listaEtiquetas){
+    let etiquetasStr = "";
+    for(i = 0; i < listaEtiquetas.length; i++){
+        let etiqueta = listaEtiquetas[i];
+        etiquetasStr += etiqueta.enlace + "," + etiqueta.coordX + "," + etiqueta.coordY + "|"
+    }
+    return etiquetasStr.substring(0, etiquetasStr.length - 1);
+}
+
+function validaEnlace(enlace) {
+    let validador = false;
+    if(/^(http|https|ftp):\/\/?.*$/i.test(enlace)){
+        validador = true;
+    }
+
+    return validador;
+}
+
+function readURL(input) {
+    let validador = true;
+    if (input.files && input.files[0]) {
+        const tipo = input.files[0].type;
+        if (tipo == "image/jpg" || tipo == "image/jpeg" || tipo == "image/png") {
+            let reader = new FileReader();
+            reader.onload = function (e) {
+                $('#visor-imagen').attr('src', e.target.result);
+                $('#visor-imagen').show();
+            }
+            reader.readAsDataURL(input.files[0]);
+            validador = true;
+        } else {
+            validador = false;
+        }
+    }
+
+    return validador;
+}
+
+function getCoords(event, element) {
+    const clickLeft = event.clientX;
+    const elementLeft = element.offsetLeft;
+    let x = (clickLeft - elementLeft) / element.width;
+    x = (Math.round(x * 1000) / 1000) - 0.02;
+
+    const clickTop = event.clientY + $(window).scrollTop() + 5;
+    const elementTop = element.offsetTop;
+    let y = (clickTop - elementTop) / element.height;
+    y = (Math.round(y * 1000) / 1000) - 0.02;
+
+    return [x, y];
+}
+
+function crearEtiqueta(coords) {
+    const x = coords[0] * 100;
+    const y = coords[1] * 100;
+    const etiquetaID = "#etiqueta" + contadorEtiqueta;
+    $("#container-visor").append('<div class="etiqueta" id="etiqueta'
+        + contadorEtiqueta + '"><div>');
+    $(etiquetaID).css("left", x + '%');
+    $(etiquetaID).css("top", y + '%');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/youRoom-ISPP/youRoom/pullRequest?prid=7157942)
```suggestion
    $(etiquetaID).css("top", y + "%");
```